src/setup.ts

```diff
@@ -33,13 +33,15 @@ async function setup(): Promise<void> {
 
         // download Notation CLI and validate checksum
         const downloadURL = getNotationDownloadURL(version, notation_url);
-        console.log(`Downloading Notation CLI from ${downloadURL}`);
+        console.log(`downloading Notation CLI from ${downloadURL}`);
         const pathToTarball: string = await tc.downloadTool(downloadURL);
-        if (notation_url) {
-            await validateCheckSum(pathToTarball, notation_checksum);
-        } else {
-            await validateCheckSum(pathToTarball, getNotationCheckSum(version)); 
+        console.log("downloading Notation CLI completed")
+        const sha256 = await hash(pathToTarball);
+        const groundTruth = notation_url ? notation_checksum : getNotationCheckSum(version);
+        if (sha256 !== groundTruth) {
+            throw new Error(`checksum of downloaded Notation CLI ${sha256} does not match ground truth ${groundTruth}`);
```

Comment:
nit: the term `ground truth` is not common in crypto. Maybe you can reword a bit.

Reply:
Sure. Updated to `expectedCheckSum`.